processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go

```diff
@@ -71,5 +77,63 @@ func (d *Detector) Detect(ctx context.Context) (pdata.Resource, error) {
 	attr.InsertString(conventions.AttributeHostType, meta.InstanceType)
 	attr.InsertString(conventions.AttributeHostName, hostname)
 
+	tags, err := connectAndFetchEc2Tags(meta.Region, meta.InstanceID, d.cfg)
+	if err != nil {
+		return res, fmt.Errorf("failed fetching ec2 instance tags: %w", err)
+	}
+
+	for key, val := range tags {
+		attr.InsertString(tagPrefix + key, val)
+	}
+
 	return res, nil
 }
+
+func connectAndFetchEc2Tags(region string, instanceID string, cfg Config) (map[string]string, error) {
+	if(!cfg.AddAllTags && len(cfg.TagsToAdd) == 0){
+		return nil, nil
+	}
+
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region)},
+	)
+	if err != nil {
+		return nil, err
+	}
+	if _, err = sess.Config.Credentials.Get(); err != nil {
+		return nil, err
+	}
+	e := ec2.New(sess)
+
+	return fetchEC2Tags(e, instanceID, cfg)
+}
+
+func fetchEC2Tags(e ec2iface.EC2API, instanceID string, cfg Config) (map[string]string, error) {
+	ec2Tags, err := e.DescribeTags(&ec2.DescribeTagsInput{
+		Filters: []*ec2.Filter{{
+			Name: aws.String("resource-id"),
+			Values: []*string{
+				aws.String(instanceID),
+			},
+		}},
+	})
+	if err != nil {
+		return nil, err
+	}
+	tags := make(map[string]string)
+	for _, tag := range ec2Tags.Tags {
+		if cfg.AddAllTags || contains(cfg.TagsToAdd, *tag.Key) {
```

Comment:
Maybe add a TODO to convert the search into a map lookup. We can send it in a follow-up PR.